### DISCUSSION TITLE
How to build a pipeline for machine learning at YASA API?

### DISCUSSION THREAD

**Discussion by PhD-GOAT** [raphaelvallat/yasa#61] [Q&A] [answered]
First, the PSG EDF is converted into a wav file for each signal for convenience of use. The converted wav file is then divided into 30-second intervals, which is the sleep stage reading cycle, and classified by reading result file provided with the data. Finally, the divided files are made into spectrogram images of wav files, and the brain waves are used as band-pass filters to extract characteristics by band.

**Comment by raphaelvallat** [comment, marked as answer]
@PhD-GOAT I don't think I understand the question. Please refer to the [eLife publication](https://elifesciences.org/articles/70092) for a description of the automatic sleep staging pipeline, and/or this [example notebook](https://github.com/raphaelvallat/yasa/blob/master/notebooks/14_automatic_sleep_staging.ipynb). Note that I do not think that the conversion to wav format is required, instead you can just directly load your EDF file into Python using the MNE library: [mne.io.read_raw_edf](https://mne.tools/stable/generated/mne.io.read_raw_edf.html).

**Reply by PhD-GOAT** [reply]
@raphaelvallat

Thank you for a good example. By the way, I want to build machine learning for the diagnosis of sleep diseases such as sleep oversleeping and narcolepsy. I referred to another paper and made it into a spectogram. So I'm going to make a wav file and machine learning.
